src/irmin-pack/unix/inode.ml

```diff
@@ -0,0 +1,29 @@
+open Import
+include Irmin_pack.Inode
+
+module Make_persistent
+    (H : Irmin.Hash.S)
+    (Node : Irmin.Node.Generic_key.S
+              with type hash = H.t
+               and type contents_key = H.t Pack_key.t
+               and type node_key = H.t Pack_key.t)
+    (Inter : Internal
+               with type hash = H.t
+                and type key = H.t Pack_key.t
+                and type Val.metadata = Node.metadata
+                and type Val.step = Node.step)
+    (CA : Pack_store.Maker
+            with type hash = H.t
+             and type index := Pack_index.Make(H).t) =
+struct
+  module Persistent_pack = CA.Make (Inter.Raw)
+  module Pack = Persistent_pack
+  module XKey = Pack_key.Make (H)
+  include Make (H) (XKey) (Node) (Inter) (Pack)
+
+  let v = Pack.v
+  let sync = Pack.sync
+  let integrity_check = Pack.integrity_check
+  let clear_caches = Pack.clear_caches
+  let integrity_check_inodes _ _ = (* fixme *) assert false
```

Comment:
This needs to be fixed. I didn't manage to make the typechecker happy moving the code from irmin-pack/inode.ml

Reply:
This seem to fix the problem
```
diff --git a/src/irmin-pack/inode_intf.ml b/src/irmin-pack/inode_intf.ml
index 7e54a7643c..f05f230bd5 100644
--- a/src/irmin-pack/inode_intf.ml
+++ b/src/irmin-pack/inode_intf.ml
@@ -60,6 +60,8 @@ module type S = sig
        and type Portable.hash := hash

   val decode_bin_length : string -> int -> int
+
+  val integrity_check_inodes : [ `Read ] t -> key -> (unit, string) result Lwt.t
 end

 module type Persistent = sig
diff --git a/src/irmin-pack/unix/inode.ml b/src/irmin-pack/unix/inode.ml
index 09d90a91f4..1bdcbd0d72 100644
--- a/src/irmin-pack/unix/inode.ml
+++ b/src/irmin-pack/unix/inode.ml
@@ -25,5 +25,5 @@ struct
   let sync = Pack.sync
   let integrity_check = Pack.integrity_check
   let clear_caches = Pack.clear_caches
-  let integrity_check_inodes _ _ = (* fixme *) assert false
 end
```

Reply:
Thanks